File: modules/yandex-map/android/src/main/java/expo/modules/yandexmap/modules/PolylineModule.kt
package expo.modules.yandexmap.modules

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.yandexmap.model.Coordinate
import expo.modules.yandexmap.view.PolylineView

class PolylineModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("Polyline")

    View(PolylineView::class) {
      Prop("points") { view: PolylineView, points: List<Coordinate> ->
        view.setPointsValue(points)
      }

      Prop("strokeColor") { view: PolylineView, color: String? ->
        view.setStrokeColorValue(color)
      }

      Prop("strokeWidth") { view: PolylineView, width: Double? ->
        view.setStrokeWidthValue(width)
      }

      Prop("outlineColor") { view: PolylineView, color: String? ->
        view.setOutlineColorValue(color)
      }

      Prop("outlineWidth") { view: PolylineView, width: Double? ->
        view.setOutlineWidthValue(width)
      }
    }
  }
}

File: modules/yandex-map/ios/PolylineModule.swift
import ExpoModulesCore

public class PolylineModule: Module {
    public func definition() -> ModuleDefinition {
        Name("Polyline")
        
        View(PolylineView.self) {
            Prop("points") { (view: PolylineView, points: Array<Coordinate>) in
                view.setPointsValue(points)
            }
            
            Prop("strokeColor") { (view: PolylineView, color: String) in
                view.setStrokeColorValue(color)
            }
            
            Prop("strokeWidth") { (view: PolylineView, width: Float) in
                view.setStrokeWidthValue(width)
            }
            
            Prop("outlineColor") { (view: PolylineView, color: String) in
                view.setOutlineColorValue(color)
            }
            
            Prop("outlineWidth") { (view: PolylineView, width: Float) in
                view.setOutlineWidthValue(width)
            }
        }
    }
}

